src/mapml.js

```diff
@@ -989,9 +1132,11 @@ M.MapMLLayer = L.Layer.extend({
       }
     },
     onRemove: function (map) {
-        L.DomUtil.remove(this._container);      
+        L.DomUtil.remove(this._container);
+        if(this._staticTileLayer){
```

Comment:
This is conditional, what it replaces wasn't.  I'm not sure why it wasn't, nor indeed at this moment why some are conditional others not.

Reply:
More a question of my own logic actually.

Reply:
The new staticTileLayer behaves similar to templatedLayer, where it is only added to the map when static tiles exist in the html file, the others (mapmlvectors, imageLayer) get added to every mapml layer container regardless of whether it's needed or not (not sure why this is the behavior, probably because in those cases you don't know if you need those layers or not prior to creating them). Since staticTileLayer is added conditionally, it is removed only if it exists.

This was actually a change in PR #106 , the difference is now the layer isn't set to null once it's removed (not needed to do anymore).